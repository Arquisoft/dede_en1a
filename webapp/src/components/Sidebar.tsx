import React from 'react'
import { Offcanvas } from 'react-bootstrap';
import Cart from './cart/Cart';

type Props = {
    handleClose: (state: boolean) => void
}

const Sidebar = ({ handleClose}: Props) => {
    return (
        <Offcanvas
            show={true}
            onHide={() => handleClose(false)}
            placement='end' >
            <Offcanvas.Header closeButton>
                <Offcanvas.Title>Your cart</Offcanvas.Title>
            </Offcanvas.Header>
            <Offcanvas.Body>
                <Cart />
            </Offcanvas.Body>
        </Offcanvas>
    )
}

export default Sidebar;