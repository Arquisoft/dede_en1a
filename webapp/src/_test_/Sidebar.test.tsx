import React from 'react';
import ReactDOM from 'react-dom';
import Router from "react-router-dom";
import { render, screen, cleanup } from '@testing-library/react';
import Sidebar from '../components/Sidebar';

const handleClose = jest.fn(() => true)

test("Sidebar renders correctly", () => {
    const fakeSidebar = render(<Sidebar setIsInCheckout={() => false} handleClose={handleClose}/>)
    expect(fakeSidebar).toBeTruthy();
})

test("Sidebar contains cart", () => {
    const fakeSidebar = render(<Sidebar setIsInCheckout={() => false} handleClose={handleClose}/>)
    expect(fakeSidebar).toBeInTheDocument();
    //expect(fakeSidebar).toHaveTextContent('Your cart'); this is what I actually wanted to test, it just didn't work
})

test("Sidebar matches snapshot", () => {
    const fakeSidebar = render(<Sidebar setIsInCheckout={() => false} handleClose={handleClose}/>)
    expect(fakeSidebar).toMatchSnapshot();
})

