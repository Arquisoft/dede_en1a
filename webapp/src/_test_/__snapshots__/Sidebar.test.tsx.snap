// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="modal-open"
    data-rr-ui-modal-open=""
    style="padding-right: 1024px;"
  >
    <div />
    <div
      class="fade offcanvas-backdrop show"
    />
    <div
      aria-modal="true"
      class="offcanvas offcanvas-end offcanvas-toggling show"
      role="dialog"
      style="visibility: visible;"
      tabindex="-1"
    >
      <div
        class="offcanvas-header"
      >
        <div
          class="offcanvas-title h5"
        >
          Your cart
        </div>
        <button
          aria-label="Close"
          class="btn-close"
          type="button"
        />
      </div>
      <div
        class="offcanvas-body"
      >
        <p>
          <b>
            Cart is empty
          </b>
        </p>
      </div>
    </div>
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
